---
title: Microbiome analysis complements fecal immunochemical test for detection of colonic lesions #must be 90 characters or less  (including spaces)
csl: nature.csl
output:
  word_document:
    reference_docx: manuscript_format.docx
bibliography: references.bibtex
---
  
  
  
  
  
Nielson T. Baxter^1^, Mack T. Ruffin IV^2^, Mary A.M. Rogers^3^, and Patrick D. Schloss^1\*^
  
  
  
^1^Department of Microbiology and Immunology, University of Michigan, Ann Arbor, Michigan.  
^2^Department of Family Medicine, University of Michigan, Ann Arbor, Michigan.  
^3^Department of Internal Medicine, University of Michigan, Ann Arbor, Michigan.  
^\*^Correspondence to: pschloss@umich.edu


```{r startup, echo=F, message=F, warning=F}
deps = c("pROC","randomForest","AUCRF","knitr","rmarkdown");
for (dep in deps){
  if (dep %in% installed.packages()[,"Package"] == FALSE){
    install.packages(as.character(dep), quiet=TRUE);
  }
  library(dep, verbose=FALSE, character.only=TRUE)
}

meta <- read.delim('data/metadata.tsv', header=T, sep='\t')
shared <- read.delim('data/glne007.final.an.unique_list.0.03.subsample.0.03.filter.shared', header=T, sep='\t')

all_data <- merge(meta, shared, by.x='sample', by.y='Group')
all_data$lesion <- factor(NA, levels=c(0,1))
all_data$lesion[all_data$dx=='adenoma' | all_data$dx=='cancer'] <- 1
all_data$lesion[all_data$dx=='normal'] <- 0
```


```{r lesion_ROC_FIT, warning=F, results='hide', echo=F, fig.width=7, fig.height=7, cache=T}
les_data <- all_data[,c('lesion','fit_result',colnames(all_data)[grep('Otu[0123456789]', colnames(all_data))])]

#generate lesion vs normal model
set.seed(53015)
les_model <- AUCRF(lesion ~ ., data=les_data, ntree=500, pdel=0.05)
pdf('results/figures/supp1_aucrf.pdf', height=5, width=5)
plot(les_model, main='AUC-RF Model: Microbiome + FIT')
dev.off()
opt_model <- les_model$RFopt
all_probs <- predict(opt_model, type='prob')
les_roc <- roc(les_data$lesion~all_probs[,2])
les_fit_roc <- roc(les_data$lesion~les_data$fit_result)

#test model on cancer
canc_data <- les_data[all_data$dx!='adenoma',]
canc_probs <- all_probs[all_data$dx!='adenoma',]
canc_roc <- roc(canc_data$lesion ~ canc_probs[,2])
canc_fit_roc <- roc(canc_data$lesion ~ canc_data$fit_result)

ade_data <- les_data[all_data$dx!='cancer',]
ade_probs <- all_probs[all_data$dx!='cancer',]
ade_roc <-  roc(ade_data$lesion ~ ade_probs[,2])
ade_fit_roc <- roc(ade_data$lesion ~ ade_data$fit_result)

par(mar=c(4.5,4.5,1,1))
pdf(file='results/figures/fig1_ROCcurve.pdf', height=6, width=6)
plot(c(1,0),c(0,1), type='l', lty=2, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(les_roc, col='purple2', lwd=2, add=T)
plot(canc_roc, col='red', lwd=2, add=T)
plot(ade_roc, col='blue', lwd=2, add=T)
plot(ade_fit_roc, col='blue', lwd=2, add=T, lty=2)
plot(canc_fit_roc, col='red', lwd=2, add=T, lty=2)
plot(les_fit_roc, col='purple2', lwd=2, add=T, lty=2)
legend('bottomright', legend=c(sprintf('Lesion: %.3g',les_roc$auc),
                               sprintf('Les. FIT: %.3g',les_fit_roc$auc),
                               sprintf('Cancer: %.3g',canc_roc$auc),
                               sprintf('Canc. FIT: %.3g',canc_fit_roc$auc),
                               sprintf('Adenoma: %.3g',ade_roc$auc),
                               sprintf('Ade. FIT: %.3g',ade_fit_roc$auc)),lty=c(1,2,1,2,1,2), lwd=2, col=c('purple2','purple2','red','red','blue','blue'), bty='n')

points(coords(canc_fit_roc, x=100, input='threshold', ret='specificity'),coords(canc_fit_roc, x=100, input='threshold', ret='sensitivity'), col='red', cex=1.5, pch=10)
points(coords(les_fit_roc, x=100, input='threshold', ret='specificity'),coords(les_fit_roc, x=100, input='threshold', ret='sensitivity'), col='purple', cex=1.5, pch=10)
points(coords(ade_fit_roc, x=100, input='threshold', ret='specificity'),coords(ade_fit_roc, x=100, input='threshold', ret='sensitivity'), col='blue', cex=1.5, pch=10)

cutoff <- coords(roc=les_roc, x='best', best.method='y', ret=c('threshold'))


points(coords(canc_roc, x=cutoff, input='threshold', ret='specificity'),coords(canc_roc, x=cutoff, input='threshold', ret='sensitivity'), col='red', cex=1.5, pch=16)
points(coords(les_roc, x=cutoff, input='threshold', ret='specificity'),coords(les_roc, x=cutoff, input='threshold', ret='sensitivity'), col='purple', cex=1.5, pch=16)
points(coords(ade_roc, x=cutoff, input='threshold', ret='specificity'),coords(ade_roc, x=cutoff, input='threshold', ret='sensitivity'), col='blue', cex=1.5, pch=16)

cat('RF model vs FIT alone\n','Lesion vs Normal: p=', roc.test(les_roc, les_fit_roc)$p.value, '\nAdenoma vs Normal: p=',roc.test(ade_roc, ade_fit_roc)$p.value,'\nCancer vs Normal: p=',roc.test(canc_roc, canc_fit_roc)$p.value)
dev.off()

```

```{r les_OTUs, warning=F, echo=F, fig.width=7, fig.height=12, results='hide'}
les_otus <- les_model$Xopt
les_otus <- les_otus[les_otus!='fit_result']
les_otus <- rev(les_otus)

tax <- read.delim('data/glne007.0.03.trim.tax', sep='\t', header=T, row.names=1)
les_tax <- paste(tax[les_otus, 'Classification'], '\n(', tax[les_otus, 'OTU'], ')', sep='')

norm_abunds <- all_data[all_data$dx=='normal', les_otus]/10000 + 1e-4
les_abunds <- all_data[all_data$dx!='normal', les_otus]/10000 + 1e-4

pdf('results/figures/supp2_modelOTUstripchart.pdf', width=7, height=12)
par(mar=c(5, 10, 0.5, 0.5))
plot(1, type="n", ylim=c(0,length(les_otus)*2), xlim=c(1e-4,1), log="x", ylab="", xlab="Relative Abundance (%)", xaxt="n", yaxt="n")
index <- 1
for(i in les_otus){
  stripchart(at=index-0.35, jitter(norm_abunds[,i], amount=1e-5), pch=21, bg="grey90", method="jitter", jitter=0.2, add=T, cex=1)
  stripchart(at=index+0.35, jitter(les_abunds[,i], amount=1e-5), pch=21, bg="grey30", method="jitter", jitter=0.2, add=T, cex=1)
  segments(mean(norm_abunds[,i]),index-0.7,mean(norm_abunds[,i]),index, lwd=3)
  segments(mean(les_abunds[,i]),index+0.7,mean(les_abunds[,i]),index, lwd=3)
  index <- index + 2
}
axis(2, at=seq(1,index-2,2), labels=les_tax, las=1, tick=F, cex.axis=1)
axis(1, at=c(1e-4, 1e-3, 1e-2, 1e-1, 1), label=c("0", "0.1", "1", "10", "100"))
legend('topright', legend=c("Lesion", "Normal"), pch=c(21, 21), pt.bg=c("grey30","grey90"))
dev.off()
```

```{r model_stripCharts, warning=F, echo=F, fig.width=8, fig.height=6, results='hide'}
pdf(file='results/figures/fig2A_stripCharts.pdf', height=5.5, width=7)
par(mfrow=c(1,2), mar=c(3, 5, 0.5, 0.5))
plot(1, type="n", xlim=c(0.5,3.5), ylim=c(0.2,1), ylab="Model Probability of Lesion", xlab="", xaxt="n")
stripchart(at=1, all_probs[all_data$dx=='normal',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=2, all_probs[all_data$dx=='adenoma',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=3, all_probs[all_data$dx=='cancer',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)
axis(1, at=c(1,2,3), labels=c('Normal','Adenoma','Cancer'))
abline(h=cutoff, lty=2)

plot(1, type="n", xlim=c(0.5,3.5), ylim=c(0.07,5000), log="y", ylab="FIT Result (ng/ml)", xlab="", xaxt="n", yaxt="n")
stripchart(at=1, jitter(all_data[all_data$dx=='normal','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=2, jitter(all_data[all_data$dx=='adenoma','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=3, jitter(all_data[all_data$dx=='cancer','fit_result']+.1, amount=0.05), vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)
axis(1, at=c(1,2,3), labels=c('Normal','Adenoma','Cancer'))
axis(2, at=c(0.1,1.1,10.1,100.1,1000.1), labels=c(0,1,10,100,1000))
abline(h=100, lty=2)
dev.off()
par(mfrow=c(1,1))

```

```{r neg_fit_stripchart, echo=F, fig.width=8, fig.height=6, results='hide'}
par(mar=c(3, 5, 0.5, 0.5))
pdf(file='results/figures/supp3_negFITstripchart.pdf', height=7, width=7) 
plot(1, type="n", xlim=c(0.5,7), ylim=c(0.2,1), ylab="Model Probability of Lesion", xlab="", xaxt="n")

stripchart(at=1, all_probs[all_data$dx=='normal' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=2, all_probs[all_data$dx=='adenoma' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=3, all_probs[all_data$dx=='cancer' & all_data$fit_result >= 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)


stripchart(at=4.5, all_probs[all_data$dx=='normal' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=5.5, all_probs[all_data$dx=='adenoma' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=6.6, all_probs[all_data$dx=='cancer' & all_data$fit_result < 100,2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

axis(1, at=c(2,5), labels=c('Positive FIT','Negative FIT'))
abline(h=cutoff, lty=2)
legend('bottomleft', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('grey','blue','red'), pch=21, bty='n', pt.cex=1.5)
dev.off()
```

```{r fit_neg_stats, echo=F}
neg_les_mod <- all_probs[all_data$dx!='normal' & all_data$fit_result < 100,2]
neg_ade_mod <- all_probs[all_data$dx=='adenoma' & all_data$fit_result < 100,2]
neg_canc_mod <- all_probs[all_data$dx=='cancer' & all_data$fit_result < 100,2]
neg_norm_mod <-  all_probs[all_data$dx=='normal' & all_data$fit_result < 100,2]

neg_les_mod_sens <- length(neg_les_mod[neg_les_mod>=cutoff]) / length(neg_les_mod)
neg_ade_mod_sens <- length(neg_ade_mod[neg_ade_mod>=cutoff]) / length(neg_ade_mod)
neg_canc_mod_sens <- length(neg_canc_mod[neg_canc_mod>=cutoff]) / length(neg_canc_mod)
neg_mod_spec <- length(neg_norm_mod[neg_norm_mod<cutoff]) / length(neg_norm_mod)
neg_mod_acc <- (length(neg_norm_mod[neg_norm_mod<cutoff]) + length(neg_les_mod[neg_les_mod>=cutoff])) / (length(neg_les_mod) + length(neg_norm_mod))

pos_les_mod <- all_probs[all_data$dx!='normal' & all_data$fit_result >= 100,2]
pos_ade_mod <- all_probs[all_data$dx=='adenoma' & all_data$fit_result >= 100,2]
pos_canc_mod <- all_probs[all_data$dx=='cancer' & all_data$fit_result >= 100,2]
pos_norm_mod <-  all_probs[all_data$dx=='normal' & all_data$fit_result >= 100,2]

pos_les_mod_sens <- length(pos_les_mod[pos_les_mod>=cutoff]) / length(pos_les_mod)
pos_ade_mod_sens <- length(pos_ade_mod[pos_ade_mod>=cutoff]) / length(pos_ade_mod)
pos_canc_mod_sens <- length(pos_canc_mod[pos_canc_mod>=cutoff]) / length(pos_canc_mod)
pos_mod_spec <- length(pos_norm_mod[pos_norm_mod<cutoff]) / length(pos_norm_mod)
pos_mod_acc <- (length(pos_norm_mod[pos_norm_mod<cutoff]) + length(pos_les_mod[pos_les_mod>=cutoff])) / (length(pos_les_mod) + length(pos_norm_mod))

output_matrix <- matrix(100*c(pos_les_mod_sens,pos_canc_mod_sens,pos_ade_mod_sens,pos_mod_spec,pos_mod_acc,
                          neg_les_mod_sens,neg_canc_mod_sens,neg_ade_mod_sens,neg_mod_spec,neg_mod_acc), byrow=T, nrow=2)
colnames(output_matrix)<-c('Lesion Sensitivity','Cancer Sensitivity','Adenoma Sensitivity','Specificity','Accuracy')
rownames(output_matrix)<-c('Positive FIT','Negative FIT')
#kable(output_matrix, digits=1)
```

```{r fit_vs_model, echo=F, fig.width=8, fig.height=7, results='hide'}
pdf('results/figures/fig3_FITvsRF.pdf', width=7, height=6)
par(mar=c(5, 5, 0.5, 0.5))
plot(1, type="n", xlim=c(0.1,10000), ylim=c(0.2,1), log='x',ylab="Model Probability of Lesion", xlab="FIT (ng/ml)", xaxt='n')

palette(c('blue','red','grey'))
points(all_data[,'fit_result'], all_probs[,2], pch=21, bg=all_data$dx)
points(jitter(rep(0.15,length(all_probs[all_data$fit_result==0,2])), factor=20), all_probs[all_data$fit_result==0,2], pch=21, bg=all_data[all_data$fit_result==0, 'dx'])

axis(1, at=c(0.15,1.1,10.1,100.1,1000.1), labels=c(0,1,10,100,1000))
abline(h=cutoff, lty=2)
abline(v=100.1, lty=2)
legend('bottomright', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('grey','blue','red'), pch=21, bty='n', pt.cex=1.5)
dev.off()
```

```{r sensitivity_barplots, echo=F, fig.width=9, fig.height=6, results='hide'}
nonAdv_probs <- all_probs[all_data$Dx_Bin=='Adenoma',2]
advAde_probs <- all_probs[all_data$Dx_Bin=='adv Adenoma',2]
s1_probs <- all_probs[all_data$stage==1,2]
s2_probs <- all_probs[all_data$stage==2,2]
s3_probs <- all_probs[all_data$stage==3,2]
s4_probs <- all_probs[all_data$stage==4,2]

nonAdv_sens <- length(nonAdv_probs[nonAdv_probs>=cutoff]) / length(nonAdv_probs)
advAde_sens <- length(advAde_probs[advAde_probs>=cutoff]) / length(advAde_probs)
s1_sens <- length(s1_probs[s1_probs>=cutoff]) / length(s1_probs)
s2_sens <- length(s2_probs[s2_probs>=cutoff]) / length(s2_probs)
s3_sens <- length(s3_probs[s3_probs>=cutoff]) / length(s3_probs)
s4_sens <- length(s4_probs[s4_probs>=cutoff]) / length(s4_probs)

nonAdv_fit <- all_data[all_data$Dx_Bin=='Adenoma','fit_result']
advAde_fit <- all_data[all_data$Dx_Bin=='adv Adenoma','fit_result']
s1_fit <- all_data[all_data$stage==1,'fit_result']
s2_fit <- all_data[all_data$stage==2,'fit_result']
s3_fit <- all_data[all_data$stage==3,'fit_result']
s4_fit <- all_data[all_data$stage==4,'fit_result']

nonAdv_fit_sens <- length(nonAdv_fit[nonAdv_fit>=100]) / length(nonAdv_fit)
advAde_fit_sens <- length(advAde_fit[advAde_fit>=100]) / length(advAde_fit)
s1_fit_sens <- length(s1_fit[s1_fit>=100]) / length(s1_fit)
s2_fit_sens <- length(s2_fit[s2_fit>=100]) / length(s2_fit)
s3_fit_sens <- length(s3_fit[s3_fit>=100]) / length(s3_fit)
s4_fit_sens <- length(s4_fit[s4_fit>=100]) / length(s4_fit)

sens_matrix <- matrix(c(nonAdv_sens,nonAdv_fit_sens,advAde_sens,advAde_fit_sens,s1_sens,s1_fit_sens,s2_sens,s2_fit_sens,s3_sens,s3_fit_sens,s4_sens,s4_fit_sens), byrow=F, nrow=2) 
rownames(sens_matrix) <- c('RF Model','FIT')

pdf('results/figures/fig2B_sensitivityBarplot.pdf', height=6, width=9)
par(mar=c(4,4,2,2))
x_labels <- c(sprintf(' \nNonadvanced\nAdenoma\n(n=%i)',length(nonAdv_probs)),sprintf('Advanced\nAdenomas\n(n=%i)',length(advAde_probs)),sprintf('Cancer\nStage I\n(n=%i)',length(s1_probs)), sprintf('Cancer\nStage II\n(n=%i)',length(s2_probs)), sprintf('Cancer\nStage III\n(n=%i)',length(s3_probs)), sprintf('Cancer\nStage IV\n(n=%i)',length(s4_probs)))
bp<-barplot(sens_matrix, beside=T, ylab='Sensitvity (%)', ylim=c(0,1.1), axisnames=F, col=c('grey25','grey90'),axis.lty=1, yaxt='n')
axis(2, at=seq(0,1,0.1), labels=seq(0,100,10))
mtext(text=x_labels, side=1, at=c(2,5,8,11,14,17), line=2)
legend('topleft', legend=c('RF Model','FIT (100ng/ml cutoff)'), col=c('grey25','grey90'), pch=15, bty='n', pt.cex=2)


#McNemar test
getMcnemarP <- function(model, fit){
  square <- matrix(c(length(model[model>=cutoff & fit >=100]),
                     length(model[model>=cutoff & fit<100]),
                     length(model[model<cutoff & fit >=100]),
                     length(model[model<cutoff & fit <100])), ncol=2, byrow=T)
  return(mcnemar.test(square)$p.value)
  }


text(2,sens_matrix[1,1]+0.05, sprintf('p=%.1g',getMcnemarP(nonAdv_probs, nonAdv_fit)))
text(5,sens_matrix[1,2]+0.05, sprintf('p=%.1g',getMcnemarP(advAde_probs, advAde_fit)))
text(8,sens_matrix[1,3]+0.05, sprintf('p=%.1g',getMcnemarP(s1_probs, s1_fit)))
text(11,sens_matrix[1,4]+0.05, sprintf('p=%.1g',getMcnemarP(s2_probs, s2_fit)))
text(14,sens_matrix[1,5]+0.05, sprintf('p=%.1g',getMcnemarP(s3_probs, s3_fit)))
text(17,sens_matrix[1,6]+0.05, sprintf('p=%.1g',getMcnemarP(s4_probs, s4_fit)))


dev.off()

```

```{r stats_table, echo=F, cache=T}

#Numbers of each diagnosis by colonoscopy
les_mod <- all_probs[all_data$dx!='normal',2]
advAde_mod <- all_probs[all_data$Dx_Bin=='adv Adenoma',2]
nonAdv_mod <- all_probs[all_data$Dx_Bin=='Adenoma',2]
ade_mod <- all_probs[all_data$dx=='adenoma',2]
canc_mod <- all_probs[all_data$dx=='cancer',2]
norm_mod <- all_probs[all_data$dx=='normal',2]
num_les <- length(les_mod)
num_advAde <- length(advAde_mod)
num_nonAdv <- length(nonAdv_mod)
num_ade <- length(ade_mod)
num_canc <- length(canc_mod)
num_norm <- length(norm_mod)

#Numbers of diagnoses observed by our model
les_obs <- length(les_mod[les_mod>=cutoff])
advAde_obs <- length(advAde_mod[advAde_mod>=cutoff])
nonAdv_obs <- length(nonAdv_mod[nonAdv_mod>=cutoff])
ade_obs <- length(ade_mod[ade_mod>=cutoff])
canc_obs <- length(canc_mod[canc_mod>=cutoff])
norm_obs <- length(norm_mod[norm_mod<cutoff])

# sensitivity and 95% CI calculations for model
les_sens_ci <- ci.coords(les_roc, x=cutoff, input='threshold', ret='sensitivity')
les_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*les_sens_ci[2],100*les_sens_ci[1],100*les_sens_ci[3])

advAde_roc <- roc(les_data$lesion[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_probs[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal',2])
nonAdv_roc <- roc(les_data$lesion[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_probs[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal',2])

advAde_sens_ci<-ci.coords(advAde_roc, x=cutoff, input='threshold', ret='sensitivity')
advAde_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*advAde_sens_ci[2],100*advAde_sens_ci[1],100*advAde_sens_ci[3])

nonAdv_sens_ci<-ci.coords(nonAdv_roc, x=cutoff, input='threshold', ret='sensitivity')
nonAdv_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*nonAdv_sens_ci[2],100*nonAdv_sens_ci[1],100*nonAdv_sens_ci[3])

ade_sens_ci<-ci.coords(ade_roc, x=cutoff, input='threshold', ret='sensitivity')
ade_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*ade_sens_ci[2],100*ade_sens_ci[1],100*ade_sens_ci[3])

canc_sens_ci<-ci.coords(canc_roc, x=cutoff, input='threshold', ret='sensitivity')
canc_mod_sens <- sprintf('%.3g (%.3g-%.3g)',100*canc_sens_ci[2],100*canc_sens_ci[1],100*canc_sens_ci[3])

mod_spec_ci<-ci.coords(les_roc, x=cutoff, input='threshold', ret='specificity')
mod_spec <- sprintf('%.3g (%.3g-%.3g)',100*mod_spec_ci[2],100*mod_spec_ci[1],100*mod_spec_ci[3])


#Numbers of diagnoses observed by FIT
les_fit <- all_data[all_data$dx!='normal','fit_result']
advAde_fit <- all_data[all_data$Dx_Bin=='adv Adenoma','fit_result']
nonAdv_fit <- all_data[all_data$Dx_Bin=='Adenoma','fit_result']
ade_fit <- all_data[all_data$dx=='adenoma','fit_result']
canc_fit <- all_data[all_data$dx=='cancer','fit_result']
norm_fit <- all_data[all_data$dx=='normal','fit_result']
les_fit_obs <- length(les_fit[les_fit>=100])
advAde_fit_obs <- length(advAde_fit[advAde_fit>=100])
nonAdv_fit_obs <- length(nonAdv_fit[nonAdv_fit>=100])
ade_fit_obs <- length(ade_fit[ade_fit>=100])
canc_fit_obs <- length(canc_fit[canc_fit>=100])
norm_fit_obs <- length(norm_fit[norm_fit<100])

# sensitivity and 95% CI calculations for FIT
les_fit_sens_ci <- ci.coords(les_fit_roc, x=100, input='threshold', ret='sensitivity')
les_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*les_fit_sens_ci[2],100*les_fit_sens_ci[1],100*les_fit_sens_ci[3])

advAde_fit_roc <- roc(les_data$lesion[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_data[all_data$Dx_Bin=='adv Adenoma' | all_data$Dx_Bin=='Normal', 'fit_result'])
nonAdv_fit_roc <- roc(les_data$lesion[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal'] ~ all_data[all_data$Dx_Bin=='Adenoma' | all_data$Dx_Bin=='Normal', 'fit_result'])

advAde_fit_sens_ci<-ci.coords(advAde_fit_roc, x=100, input='threshold', ret='sensitivity')
advAde_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*advAde_fit_sens_ci[2],100*advAde_fit_sens_ci[1],100*advAde_fit_sens_ci[3])

nonAdv_fit_sens_ci<-ci.coords(nonAdv_fit_roc, x=100, input='threshold', ret='sensitivity')
nonAdv_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*nonAdv_fit_sens_ci[2],100*nonAdv_fit_sens_ci[1],100*nonAdv_fit_sens_ci[3])

ade_fit_sens_ci<-ci.coords(ade_fit_roc, x=100, input='threshold', ret='sensitivity')
ade_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*ade_fit_sens_ci[2],100*ade_fit_sens_ci[1],100*ade_fit_sens_ci[3])

canc_fit_sens_ci<-ci.coords(canc_fit_roc, x=100, input='threshold', ret='sensitivity')
canc_fit_sens <- sprintf('%.3g (%.3g-%.3g)',100*canc_fit_sens_ci[2],100*canc_fit_sens_ci[1],100*canc_fit_sens_ci[3])

fit_spec_ci<-ci.coords(les_fit_roc, x=100, input='threshold', ret='specificity')
fit_spec <- sprintf('%.3g (%.3g-%.3g)',100*fit_spec_ci[2],100*fit_spec_ci[1],100*fit_spec_ci[3])

#Build Output table

stats_table <- matrix(NA, ncol=6, nrow=8)
colnames(stats_table) <- c('Most Advanced Finding','Colonoscopy','RF Model','','FIT','')
stats_table[1,] <- c('','','Positive Results','Sensitivity (95% CI)','Positive Results','Sensitivity (95% CI)')
stats_table[2,] <- c('','no.','no.','%','no.','%')
stats_table[3,] <- c('Cancer', num_canc, canc_obs, canc_mod_sens, canc_fit_obs, canc_fit_sens)
stats_table[4,] <- c('Advanced Adenoma', num_advAde, advAde_obs, advAde_mod_sens, advAde_fit_obs, advAde_fit_sens)
stats_table[5,] <- c('Nonadvanced Adenoma', num_nonAdv, nonAdv_obs, nonAdv_mod_sens, nonAdv_fit_obs, nonAdv_fit_sens)
stats_table[6,] <- c('Any Lesion', num_les, les_obs, les_mod_sens, les_fit_obs, les_fit_sens)
stats_table[7,] <- c('','','Negative Results','Specificity (95% CI)','Negative Results','Specificity (95% CI)')
stats_table[8,] <- c('Negative Colonosopies', num_norm, norm_obs, mod_spec, norm_fit_obs, fit_spec)

write(kable(stats_table), file='results/tables/table1.md')
#render('results/tables/table.md', output_format='pdf_document', output_file='table1.pdf')

```

```{r predictive_values, echo=F, results='hide'}
getPV <- function(sens, spec, prev){
  ppv <- signif((sens*prev) / (sens*prev + (1-spec)*(1-prev)), 4)
    
  npv <- signif((spec*(1-prev)) / ( (1-sens)*prev + spec*(1-prev) ),4) 
  
  return(list(PPV=ppv, NPV=npv))
}

advAde_pv <- getPV(advAde_sens_ci[2],mod_spec_ci[2],0.057)
nonAdv_pv <- getPV(nonAdv_sens_ci[2],mod_spec_ci[2],0.177)
canc_pv <- getPV(canc_sens_ci[2],mod_spec_ci[2],0.003)
les_pv<- getPV(les_sens_ci[2],mod_spec_ci[2],0.302)

advAde_fit_pv <- getPV(advAde_fit_sens_ci[2],fit_spec_ci[2],0.057)
nonAdv_fit_pv <- getPV(nonAdv_fit_sens_ci[2],fit_spec_ci[2],0.177)
canc_fit_pv <- getPV(canc_fit_sens_ci[2],fit_spec_ci[2],0.003)
les_fit_pv<- getPV(les_fit_sens_ci[2],fit_spec_ci[2],0.302)

pv_table <- matrix(NA, ncol=5, nrow=5)
colnames(pv_table) <- c('Prevalence','PPV','','NPV','')
rownames(pv_table) <- c('','Cancer','Adv. Adenoma','Nonadv. Ade.','All Lesions')
pv_table[1,] <- c('','RF Model','FIT','RF Model','FIT')
pv_table[2,] <- c('0.3%', canc_pv$PPV, canc_fit_pv$PPV, canc_pv$NPV, canc_fit_pv$NPV)
pv_table[3,] <- c('5.7%', advAde_pv$PPV, advAde_fit_pv$PPV, advAde_pv$NPV, advAde_fit_pv$NPV)
pv_table[4,] <- c('17.7%', nonAdv_pv$PPV, nonAdv_fit_pv$PPV, nonAdv_pv$NPV, nonAdv_fit_pv$NPV)
pv_table[5,] <- c('30.2%', les_pv$PPV, les_fit_pv$PPV, les_pv$NPV, les_fit_pv$NPV)

write(kable(stats_table), file='results/tables/supp_table2.md')
#render('results/tables/table.md', output_format='pdf_document', output_file='supp_table2.pdf')

```

```{r male_vs_female, echo=F, results='hide', fig.width=9, fig.height=6}
m_roc <- roc(all_data$lesion[all_data$Gender == 'm'] ~ all_probs[all_data$Gender == 'm',2])
f_roc <- roc(all_data$lesion[all_data$Gender == 'f'] ~ all_probs[all_data$Gender == 'f',2])
gender_pval <- roc.test(m_roc,f_roc)$p.value

layout(matrix(c(1,2),nrow=1), widths=c(8,5))
par(mar=c(4.5,4.5,1,1))
pdf('results/figures/supp4_maleFemale.pdf', height=6, width=9)

plot(c(1,0),c(0,1), type='l', lty=2, xlim=c(1.01,0), ylim=c(-0.01,1.01), xaxs='i', yaxs='i', ylab='Sensitivity', xlab='Specificity')
plot(les_roc, col='purple', lwd=2, add=T)
plot(m_roc, col='blue', lwd=2, add=T)
plot(f_roc, col='red', lwd=2, add=T)
points(coords(les_roc, x=cutoff, input='threshold', ret='specificity'),coords(les_roc, x=cutoff, input='threshold', ret='sensitivity'), col='purple', cex=1.5, pch=16)
points(coords(m_roc, x=cutoff, input='threshold', ret='specificity'),coords(m_roc, x=cutoff, input='threshold', ret='sensitivity'), col='blue', cex=1.5, pch=16)
points(coords(f_roc, x=cutoff, input='threshold', ret='specificity'),coords(f_roc, x=cutoff, input='threshold', ret='sensitivity'), col='red', cex=1.5, pch=16)

legend('bottomright', legend=c(sprintf('All Subjects: %.3g',les_roc$auc),
                               sprintf('Males: %.3g',m_roc$auc),
                               sprintf('Females: %.3g',f_roc$auc),
                               sprintf('Male vs Female: p=%.3g',gender_pval)
                               ),lty=c(1,1,1,0), lwd=2, col=c('purple','blue','red'), bty='n')


plot(1, type="n", xlim=c(0.5,7), ylim=c(0,1), ylab="Model Probability of Lesion", xlab="", xaxt="n")
stripchart(at=1, all_probs[all_data$dx=='normal' & all_data$Gender == 'm', 2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=2, all_probs[all_data$dx=='adenoma' & all_data$Gender == 'm',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=3, all_probs[all_data$dx=='cancer' & all_data$Gender == 'm',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

stripchart(at=4.5, all_probs[all_data$dx=='normal' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='grey', add=T)
stripchart(at=5.5, all_probs[all_data$dx=='adenoma' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='blue', add=T)
stripchart(at=6.6, all_probs[all_data$dx=='cancer' & all_data$Gender == 'f',2], vertical=T, method='jitter', jitter=0.4, pch=21, bg='red', add=T)

axis(1, at=c(2,5), labels=c('Males','Females'))
abline(h=cutoff, lty=2)
legend('bottomleft', legend=c('Normal','Adenoma','Cancer'), pt.bg=c('grey','blue','red'), pch=21, bty='n', pt.cex=1.5)
dev.off()
```

```{r gender_normalized, echo=F, fig.width=9, fig.height=6, cache=T, results='hide'}

row_nums <- 1:nrow(all_data)
pvals <- numeric()
for(i in 1:1000){
  m_norm <- sample(row_nums[all_data$Gender=='m' & all_data$dx=='normal'], 61)
  m_ade <- sample(row_nums[all_data$Gender=='m' & all_data$dx=='adenoma'], 80)
  m_canc <- sample(row_nums[all_data$Gender=='m' & all_data$dx=='cancer'], 52)
  f_norm <- sample(row_nums[all_data$Gender=='f' & all_data$dx=='normal'], 61)
  f_ade <- sample(row_nums[all_data$Gender=='f' & all_data$dx=='adenoma'], 80)
  f_canc <- sample(row_nums[all_data$Gender=='f' & all_data$dx=='cancer'], 52)

  m_rows <- c(m_norm,m_ade,m_canc)
  f_rows <- c(f_norm,f_ade,f_canc)

  m_roc <- roc(all_data$lesion[m_rows] ~ all_probs[m_rows,2])
  f_roc <- roc(all_data$lesion[f_rows] ~ all_probs[f_rows,2])

  pvals[i] <- roc.test(m_roc,f_roc)$p.value
}

#cat('Male vs Female\n % of tests with p-value < 0.05:', length(pvals[pvals<0.05])/length(pvals))

```

```{r figures, echo=F}

ROC <- 'Fig. 1'
table1 <- 'Table 1'
strip <- 'Fig. 2A'
bar <- 'Fig. 2B'
scatter <- 'Fig. 3'


aucrf_curve <- 'Extended Data Fig. 1'
otu_table <- 'Extended Data Fig. 1'
pred_values <- 'Extended Data Table 2'
fit_neg_strip <- 'Extended Data Fig. 3'
gender <- 'Extended Data Fig. 4'
OTUs <- 'Extended Data Fig. 5'


```

**Colorectal cancer is the second leading cause of death among cancers in the United States [@siegel2014colorectal]. Although individuals diagnosed early have a greater than 90% chance of survival, more than one-third of individuals do not adhere to screening recommendations partly because the standard diagnostics, colonoscopy and sigmoidocsopy, are expensive and invasive [@siegel2014colorectal;@centers2010vital;@hsia2000importance;@jones2010barriers]. Thus, there is a great need to improve the sensitivity of non-invasive tests to detect early stage cancers and adenomas. Numerous studies have demonstrated a causal link between the formation of colonic lesions and the activity of the gut microbiota in tissue culture and animal models [@zackularMbio;@kostic2013fusobacterium;@wu2009human;@arthur2012intestinal]. These findings have been complemented by studies in human populations identifying shifts in the composition of the gut microbiota associated with the progression of colorectal cancer [@wang2012structural;@chen2013decreased;@chen2012human;@shen2010molecular]. These results suggest that the gut microbiota may represent a reservoir of biomarkers that would complement existing non-invasive methods such as the widely used fecal immunochemical test (FIT). Using stool samples from 490 patients we developed a cross-validated random forest classification model that detects colonic lesions using the relative abundance of gut microbiota and the concentration of hemoglobin in stool. The microbiome-based random forest model detected 95.0% of cancers and 61.1% of adenomas while FIT detected 75.0% and 15.7%, respectively. Of the colonic lesions missed by FIT, the model detected 80.0% of cancers and 53.9% of adenomas. To date this is the largest study to characterize alterations in the microbiota associated with colorectal cancer and the first to demonstrate a method by which microbiome analysis improves upon existing screening methods. With a negative predictive value of 99.98%, our model could be used to accurately identify those patients for whom a colonoscopy is unnecessary, potentially reducing healthcare costs and complications due to invasive screening.**

CRC incidence and mortality have steadily declined in recent decades, due in large part to increased screening[@siegel2014colorectal]. Further progress is possible by increasing access to and accuracy of diagnostic tests. Although structural exams like colonoscopy and sigmoidoscopy are able to detect both cancer and adenomas, their high cost and invasive nature are barriers for many people. For example, fear, discomfort, and embarrassment are among the most cited reasons patients choose not to undergo CRC screening [@jones2010barriers]. Likewise the large disparity in screening rates between those with and without health insurance highlights the need for less expensive screen methods [@hsia2000importance;@centers2010vital;@siegel2014colorectal]. Unfortunately cheaper, less invasive stool-based tests like guaic fecal occult blood test and FIT are unable to reliably detect adenomas [@hundt2009comparative]. Thus there is need for novel screening methods that are inexpensive and capable of detecting both cancer and adenomas.

The gut microbiota, the collection of microorganisms that inhabit the gastrointestinal tract, are one potential source of biomarkers for detecting colonic lesions. Numerous studies have observed alterations in the gut bacterial communities of patients with CRC [@wang2012structural;@chen2013decreased;@chen2012human;@shen2010molecular]. Experiments in animal models have demonstrated that such alterations have the potential to accelerate tumorigenesis [@zackularMbio]. Furthermore several members of the gut microbiota have been shown to potentiate both the development and progression of CRC by a variety of mechanisms [@kostic2013fusobacterium;@wu2009human;@arthur2012intestinal]. While each of these organisms may play a role in certain cases of CRC, none of them is present in every case. Therefore no one organism is an effective biomarker on its own. 

We have perviously shown that statistical models that take into account the abundances of multiple bacteria species can be used to distinguish healthy individuals from those with colonic lesions [@zackular2014human]. In the present study we expanded upon those findings by demonstrating the potential for microbiome analysis to complement FIT for improved detection of colonic lesions. We did so using the largest patient cohort to date for studying differences in the microbiomes of patients with CRC. We also improved upon previous studies by utilizing random forest (RF), a decision tree-based machine algorithm for classification, which includes an internal cross-validation to prevent overfitting [liaw2002classification]. By incorporating both FIT and bacterial abundances into a single RF model, we were able to improve the sensitivity for adenomas and most stages of cancer compared to FIT alone.
 
First we characterized the bacterial communities of stool samples from 490 patients using 16S rRNA gene sequencing.  Of those patients, 120 had CRC, 198 had adenomas, and 172 had no colonic lesions. Additionally we tested each sample for the concentraion of occult blood using FIT. With these data we developed a RF model that would differentiate normal individuals from those with any type of colonic lesion (i.e. adenoma or carcinoma). We determined the optimal model using the AUC-RF algorithm for maximizing the area under the curve (AUC) of a RF model [@calle2011auc]. The optimal model combining FIT and the microbiota used 23 bacterial populations, or operational taxonomic units (OTUs) (`r aucrf_curve`). Of those OTUs, 14 were members of the Clostridia, including 10 from the Lachnospiraceae family (OTUs 14, 44, 8, 88, 60, 22, 9, 13, 87, 31) and 2 from the Ruminococcaceae family (OTU29, OTU11). Three OTUs were associated with the genus *Bacteroides* (OTUs 3, 7, 2). The remaining OTUs were associated with *Porphyromonas* (OTU105), *Parabacteroides* (OTU49), *Streptococcus* (OTU20), and Enterobacteriaceae (OTU28). Interestingly the majority of OTUs used in the model were enriched in normal patients, suggesting that a loss of beneficial organisms in addition to the emergence of pathogens may be indicative of CRC development. 

To determine whether microbiome analysis complements FIT, we compared the performance of the RF model to using FIT alone. The AUC for the RF model was significantly higher than FIT for distinguishing adenoma from normal (p=`r signif(roc.test(ade_roc, ade_fit_roc)$p.value, 2)`) or all lesions from normal (p=`r signif(roc.test(les_roc, les_fit_roc)$p.value, 2)`), but not cancer from normal (p=`r signif(roc.test(canc_roc, canc_fit_roc)$p.value, 2)`) (`r ROC`). Examination of the ROC curves for the two tests shows that the RF model does not outperform FIT until the specificity drops below approximately 0.9, at which point the sensitivity of the RF model greatly excedes that of FIT. 

Next we defined an optimal cutoff for the RF model using Youden's J statistic [@youden1950index]. We then compared the sensitivtity and specificity of the model at this defined cutoff (`r signif(cutoff, 3)`) to using FIT with the widely accepted cutoff of 100ng/ml of hemoglobin.  At these cutoffs the RF model detected `r signif(100*canc_sens_ci[2], 3)`% of cancers and `r signif(100*ade_sens_ci[2], 3)`% of adenomas compared to `r signif(100*canc_fit_sens_ci[2], 3)`% and `r signif(100*ade_fit_sens_ci[2], 3)`% for FIT (`r cat(table1, strip, sep=', ')`). When adenomas and cancers were pooled together, the RF model detected `r signif(100*les_sens_ci[2], 3)`% of lesions, while FIT only detected `r signif(100*les_fit_sens_ci[2], 3)`%. The RF model had significantly improved sensitivity for both advanced and non-advanced adenomas as well as most stages of cancer (`r bar`). The increased sensitivity of the RF model was accompanied by a substantial decrease in specificity (`r 100*signif(mod_spec_ci[2], 3)`%) compared to FIT (`r 100*signif(fit_spec_ci[2], 3)`%). 

To better understand the relationship between the RF model and FIT, we compared the results of the two tests for each sample (`r scatter`). All samples that tested positive by FIT also tested positive in the RF model, meaning the RF model did not miss any of the lesions that FIT was able to detect. However the RF model was able to detect `r signif(100*neg_canc_mod_sens, 3)`% of cancers and `r signif(100*neg_ade_mod_sens, 3)`% of adenomas that FIT failed to detect, while maintaining a specificity of `r signif(100*neg_mod_spec, 3)`% (`r fit_neg_strip`).

As a final metric of our model's performance we estimated the positive predictive value (PPV) and negative predictive value (NPV) by extrapolating its performance on an average-risk population using previously published values for CRC prevalence[@heitman2009prevalence] (`r pred_values`). Based on a prevalence of 0.3% for CRC, the model would have a relatively low PPV of `r 100*signif(canc_pv$PPV, 3)`%, but a high NPV of `r 100*signif(canc_pv$NPV, 4)`%. For advanced adenomas the model would have a PPV of `r 100*signif(advAde_pv$PPV, 3)`% and NPV of `r 100*signif(advAde_pv$NPV, 3)`% assuming a prevalence of 5.7%. With a prevalence of 17.7% for nonadvanced adenomas, the PPV for the model would be `r 100*signif(nonAdv_pv$PPV, 3)`% and the NPV would be `r 100*signif(nonAdv_pv$NPV, 3)`%.

Previous studies have identified differences in diagnostic test performance for certain demographic groups or for people taking certain medications [@symonds2015factors;@kapidzic2015gender;@levi2009sensitivity]. Therefore we tested whether the RF model performance differed between patient populations. The model performed significantly better for females than males (`r sprintf('p=%.2g',gender_pval)`; `r gender`). We suspected that the difference could be due to having more adenoma samples from males (n=118) than females (n=80). To correct for this we normalized samples such that male and female groups contained equal numbers of adenomas, cancers, and normal samples. We resampled 1000 times and tested for a significant difference in AUC between males and females. With this method `r 100*length(pvals[pvals<0.05])/length(pvals)`% of the iterations resulted in a p-value less than 0.05, making it unclear whether gender truly affects the performance of the model. Additionally, we found no difference in model performance according to age, BMI, NSAID usage, diabetes, smoking, or previous history of polyps (data not shown). 
 
Our findings demonstrate the potential for combining microbioime analysis with conventional stool-based tests to improve CRC detection. Using the RF algorithm made it possible to interpret FIT results in the context of the microbiome and vice versa. The RF model had significantly higher sensitivity for lesions at almost all stages of tumorigenesis. Moreover the model detected the majority of lesions that FIT was unable to detect. The shortcomings of the RF model were its lack of specificity and low PPV. In other systems, such flaws would result in potentially dangerous over treatment. However in the case of CRC, patients are already recommended to receive regular colonoscopies. Therefore the potential value of the RF model is in its high sensitivity and NPV. With an NPV of `r 100*signif(canc_pv$NPV, 4)`%, the model could be used to determine those patients for whom regular colonoscopies are unnecessary. This strategy could result in a decrease in the number of colonosopies, thereby reducing both the financial costs and potential health risks of more invasive screening methods.

The accuracy of the RF model could potentially be improved by incorporating additional biomarkers. For example several bacterial toxins are capable of exacerbating tumorigeneisis, and there is evidence that methanogenic archaea are associated with CRC [@mira2014microbial]. These microbial biomarkers could be combined with 16S rRNA sequencing for a more comprehensive analysis of the microbiome. Likewise host-associated biomakers, such as those used in sDNA tests could add to the our model [@imperiale2014multitarget]. We have shown previously that patient demographic information can strengthen microbiome-based models for distinguishing healthy patients from those with CRC [@zackular2014human]. Thus combining host and microbial biomarkers in the context of an individual's risk could further improve screening accuracy. 

**Methods Summary.** Fecal samples were collected from 490 subjects in 4 locations: Toronto (Ontario, Canada), Boston (Massachusetts, USA), Houston (Texas, USA), and Ann Arbor (Michigan, USA). Patient diagnoses were determined by colonoscopy and subsequent histopathological examination. FIT was performed using OC FIT-CHEK sampling bottles and processed using an OC-Auto Micro 80 automated system (Polymedco Inc.). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers and sequenced as described previously using an Illumina MiSeq sequencer [@miseq]. A data analysis pipeline and all necessary scripts are available at github.com/SchlossLab/Baxter_glne007Modeling_2015.

### Methods (online only) 
_**Study Design/Patient sampling.**_ Elligible patients for this study were at least 18 years old, willing to sign informed consent, able to tolerate removal of 58ml of blood, and willing to collect a stool sample. Patients were excluded if they had undergone surgery, radiation, or chemotherapy for current CRC prior to baseline samples or had inflammatory bowel disease, known hereditary nonpolyposis CRC, or familial adenomatous polyposis. Colonoscopies were performed and fecal samples were collected from subjects in 4 locations: Toronto (Ontario, Canada), Boston (Massachusetts, USA), Houston (Texas, USA), and Ann Arbor (Michigan, USA). Patient diagnoses were determined by colonoscopic examination. Lesions were biopsied and diagnosed as adenoma or cancer based on subsequent histopathological examination. Whole evacuated stool was collected from each patient into hat, packed on ice, shipped to a processing center via next day delivery, and stored at -80ËšC.
			 
_**Fecal Immunochemical Tests.**_ Fecal material for FIT was collected from frozen stool aliquots using OC FIT-CHEK sampling bottles (Polymedco Inc.) and processed using an OC-Auto Micro 80 automated system (Polymedco Inc.). Raw FIT results were used for generating ROC curves and for building RF models. Sensitivities and specificities reported for FIT are based on a cutoff of 100ng/ml.


_**16S rRNA Sequencing.**_ DNA was extracted from roughly 50mg of fecal material from each subject using the PowerSoil-htp 96 Well Soil DNA isolation kit (MO BIO Laboratories) and an epMotion 5075 automated pipetting system (Eppendorf). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers and sequenced as described previously using an Illumina MiSeq sequencer [@miseq]. The 490 samples were divided into three sequencing runs to increase sequencing depth.

_**Sequence Curation.**_ The 16S rRNA gene sequences were curated using the mothur software package, as described previously [@miseq]. Briefly, paired-end reads were merged into contigs, screened for quality, aligned to SILVA 16S rRNA sequence database, and screened for chimeras. Curated sequences were clustered in to operationall taxonomic units (OTUs) using a 97% similarity cutoff. The number of sequences in each sample was rarefied to 10,000 per sample to minimize the effects of uneven sampling.

_**Statistical Methods.**_ All statistical analyses were performed using the R software package (citation). Random Forest models were generated using the AUCRF package [@calle2011auc]. The AUC of ROC curves was compared using the method described by DeLong et al. [@delong1988comparing]. The optimal cutoff for the RF model was determined using Youden's J statistic as implemented in the pROC package in R [@youden1950index]. The sensitivities of FIT and the RF model were compared using McNemar's chi-squared test.

_**Data Availability.**_ Raw fastq files and MIMARKS file are available through the NCBI Sequence Read Archive. A data analysis pipeline and all necessary scripts are available at github.com/SchlossLab/Baxter\_glne007Modeling_2015.

### Literature cited
